### ISSUE TITLE
Plot of multidimensional node fails with ValueError

### ISSUE THREAD

**Issue by fonnesbeck** [stan-dev/pystan#268]
While `plot` seems to work for simple vector-valued nodes, multidimensional nodes currently fail. For example, for a correlation matrix `corr_matrix[2] Omega_b;` that is modeled with a `lkj_corr` distribution, I get this failure:

```
>>> fit.plot(pars=['Omega_b'])

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-34-8e77554e34ed> in <module>()
----> 1 fit.plot(pars=['Omega_b'])

stanfit4anon_model_456ec81631c7e21b1d5f3580dbc6353b_6997922924549208434.pyx in stanfit4anon_model_456ec81631c7e21b1d5f3580dbc6353b_6997922924549208434.StanFit4Model.plot (/var/folders/z_/kgsm76k94s7_jhnkfydfnd380000gn/T/tmp0k60d8b1/stanfit4anon_model_456ec81631c7e21b1d5f3580dbc6353b_6997922924549208434.cpp:10206)()

/Users/fonnescj/anaconda3/envs/dev/lib/python3.5/site-packages/pystan/plots.py in traceplot(fit, pars, **kwargs)
     17         logger.critical("matplotlib required for plotting.")
     18         raise
---> 19     return plots.traceplot(fit.extract(), pars, **kwargs)

/Users/fonnescj/anaconda3/envs/dev/lib/python3.5/site-packages/pystan/external/pymc/plots.py in traceplot(trace, vars)
     39             ax[i, 0].hist(d, bins=sqrt(d.size))
     40         else:
---> 41             kdeplot_op(ax[i, 0], d)
     42         ax[i, 0].set_title(str(v))
     43         ax[i, 1].plot(d, alpha=.35)

/Users/fonnescj/anaconda3/envs/dev/lib/python3.5/site-packages/pystan/external/pymc/plots.py in kdeplot_op(ax, data)
     58         x = np.linspace(0, 1, 100) * (u - l) + l
     59 
---> 60         ax.plot(x, density(x))
     61 
     62 

/Users/fonnescj/anaconda3/envs/dev/lib/python3.5/site-packages/scipy/stats/kde.py in evaluate(self, points)
    202                 msg = "points have dimension %s, dataset has dimension %s" % (d,
    203                     self.d)
--> 204                 raise ValueError(msg)
    205 
    206         result = zeros((m,), dtype=float)

ValueError: points have dimension 1, dataset has dimension 4000
```

The dimensionality of this node is:

```
fit.extract(pars=['Omega_b']).shape

(4000, 2, 2)
```

Running PyStan built from master on Python 3.5.2 (anaconda) and macOS 10.12.

**Comment by ariddell** [comment]
Thanks. We should do something about this. Does RStan handle this?

There was/is an attempt to build a more robust plotting library for everyone who has data like this, https://github.com/mcmcplotlib/mcmcplotlib but there hasn't been much progress. After PyStan 3 gets done (which will fix all the pickling problems for good) I may have some time to work on plots. Any help would be appreciated.

**Comment by hanyas** [comment]
Is there a way to get around this error in the meanwhile?

**Comment by ahartikainen** [comment]
See the code on recent PR. You can copy the functions if needed.

**Comment by hanyas** [comment]
Thanks! The new plots look great btw

**Comment by ahartikainen** [comment]
Fixed temporary by #474 

In future, this is fixed in ArviZ library.
